eox_nelp/api_clients/pearson_rti.py

```diff
@@ -0,0 +1,144 @@
+"""Client module for Pearson VUE RTI service.
+
+Classes:
+    PearsonRTIApiClient: Class to interact with the pearson vue Real-Time Import external service.
+"""
+import logging
+
+from django.conf import settings
+
+from eox_nelp.api_clients import AbstractSOAPClient
+from eox_nelp.api_clients.authenticators import PKCS12Authenticator
+
+try:
+    from eox_audit_model.decorators import audit_method
+except ImportError:
+    def audit_method(action):  # pylint: disable=unused-argument
+        """Identity audit_method"""
+        return lambda x: x
+
+
+LOGGER = logging.getLogger(__name__)
+
+
+class PearsonRTIApiClient(AbstractSOAPClient):
+    """This class implements the required method to consume APIs available in the Pearson VUE systems
+    for importing candidate demographic and exam authorization data.
+    """
+    authentication_class = PKCS12Authenticator
+
+    def __init__(self):
+        self.cert = getattr(settings, "PEARSON_RTI_CERT")
+        self.passphrase = getattr(settings, "PEARSON_RTI_PASSPHRASE")
+
+        super().__init__()
+
+    @property
+    def base_url(self):
+        return getattr(settings, "PEARSON_RTI_WSDL_URL")
+
+    def ping(self, payload):
+        """The Ping service allows to ping the VUE system to verify that web services are available.
+        This allows to validate the credentials(ping_database) if they are included into the payload.
+
+        Arguments:
+            payload<str>: Body xml request in string format.
+
+        Returns:
+            <Dict>: Parsed xml response to Dict format.
+        """
+        result = {
```

Comment:
Here we could build a dict with more XML representation of dict.
I think using something like.
![image](https://github.com/eduNEXT/eox-nelp/assets/51926076/9947b38b-6e7e-4d1b-a82b-d4daada8a14e)
But for that we need the XML string

Reply:
The image that you shared is a request image, that has more data but here we should talk about responses and since this is the ping I share with you the response of a ping request

``` xml
<soap:Envelope xmlns:soap="http://www.w3.org/2003/05/soap-envelope" xmlns:soapenv="http://schemas.xmlsoap.org/soap/envelope/">
    <soap:Body>
        <ping:pingResponse xmlns:ping="http://ws.pearsonvue.com/ping/schema">
            <result status="success" env="CTT!" inProduction="false"/>
        </ping:pingResponse>
    </soap:Body>
</soap:Envelope>
```

So please tell why do you think that having  keys like `soap:Envelope` or `soap:Body` will improve or affect the implementation ?

Reply:
Not but maybe storing the dict representation we could look how their API is changing or sending different keys . may be using eox-audit

Reply:
if something changes, the code won't work  on the other hand if we implement that and something changes the code will work  but  the code that depends on this will fail, so I prefer to keep a stable output for all methods and handle errors here  and also that suggestion needs an extra library  that is not needed when there is a different workaround

Reply:
ok but that output still depends on the beautiful_soup. In that case I prefer here and xml_response string instead of beatul soup. https://github.com/eduNEXT/eox-nelp/blob/e196a43694caf6a47f6e9fb8bf0c4de65b008627/eox_nelp/api_clients/pearson_rti.py#L101

Reply:
ok, but please tell me how to do that

Reply:
With the change I think that keeping the string is enough